accounting/accounting_service.bal

```diff
@@ -0,0 +1,118 @@
+import ballerina/http;
+import ballerina/log;
+import ballerina/sql;
+
+# Request body to be used when charging a consumer
+type ChargeRequest record {|
+    # The ID of the consumer to be charged  
+    int consumerId;
+    # The ID of the order associated with the charge  
+    int orderId;
+    # The amount to be charged
+    decimal orderAmount;
+|};
+
+# Response when a consumer is successfully charged
+type ConsumerCharged record {|
+    *http:Ok;
+    # The deatils of the generated bill along with the relevant HTTP links
+    record {|
+        *Bill;
+        *http:Links;
+    |} body;
+|};
+
+# Response when a bill is successfully retrieved
+type BillView record {|
+    *http:Ok;
+    # The deatils of the generated bill along with the relevant HTTP links
+    record {|
+        *Bill;
+        *http:Links;
+    |} body;
+|};
+
+# Response when a requested bill cannot be found
+type BillNotFound record {|
+    *http:NotFound;
+    # Error message
+    readonly record {} body = { 
+        "message": "Bill cannot be found." 
+    };
+|};
+
+# Represents an unexpected error
+type InternalError record {|
+   *http:InternalServerError;
+    # Error payload
+    record {| 
+        string message;
+    |} body;
+|}; 
+
+# Listener for the Accounting service
+service on new http:Listener(8083) {
+
+    # Resource function to charge a consumer for an order placed
+    #
+    # + request - Details of the charge
+    # + return - `ConsumerCharged` if the charge was successfully initiated.
+    #            `InternalError` if an unexpected error occurs
+    isolated resource function post charge(@http:Payload ChargeRequest request) returns ConsumerCharged|InternalError {
+        log:printInfo("Charge request", request = request);
+        do {
+            Bill generatedBill = check createBill(request.consumerId, request.orderId, request.orderAmount);
+            check chargeConsumer(request.consumerId, request.orderAmount);
+
+            return <ConsumerCharged>{ 
+                body:{
+                    ...generatedBill,
+                    links: getLinks(generatedBill.id)
+                } 
+            };
+        } on fail error e {
+            log:printError(e.message(), e, e.stackTrace());
```

Comment:
Shall we add a proper log message instead of `e.message()`? because error message is captured in 'error as well.

Reply:
please check other places

Reply:
Like this https://github.com/ballerina-guides/ftgo-microservices-example/pull/1/files#diff-50c381d4a3e05957c8ef6c95626772fe1f6b03b43dba8b842a67aa83f75f4ec6R149